scripts/OverlaySurvey.py

```diff
@@ -289,28 +318,48 @@ def run_survey(args):
         try:
             SIMULATION = sim.SurveySimulation(args.simGraph, args.simRoot)
         except sim.SimulationError as e:
-            print(f"Error: {e}")
+            logger.critical("%s", e)
             sys.exit(1)
 
     url = args.node
 
     peers = url + "/peers"
-    survey_request = url + "/surveytopology"
+    start_collecting = url + "/startsurveycollecting"
+    stop_collecting = url + "/stopsurveycollecting"
     survey_result = url + "/getsurveyresult"
-    stop_survey = url + "/stopsurvey"
-
-    duration = int(args.duration)
-    params = {'duration': duration}
 
-    # reset survey
-    get_request(url=stop_survey)
+    # start survey recording
+    nonce = random.randint(0, 2**32-1)
+    logger.info("Starting survey with nonce %s", nonce)
+    response = get_request(url=start_collecting, params={'nonce': nonce})
+    if response.text != util.START_SURVEY_COLLECTING_SUCCESS_TEXT:
+        logger.critical("Failed to start survey: %s", response.text)
+        sys.exit(1)
+
+    # Sleep for duration of collecting phase
+    logger.info("Sleeping for collecting phase (%i minutes)", args.duration)
+    time.sleep(args.duration * 60)
+
+    # Stop survey recording
+    logger.info("Stopping survey collecting")
+    response = get_request(url=stop_collecting)
+    if response.text != util.STOP_SURVEY_COLLECTING_SUCCESS_TEXT:
+        logger.critical("Failed to stop survey: %s", response.text)
+        sys.exit(1)
+
+    # Allow time for stop message to propagate
+    sleep_time = 30
```

Comment:
maybe increase it to a minute just in case we're dealing with slow nodes on the edges of the network?

Reply:
Good point. Increased to 1 minute in 1a36c486dca5149c1f619ea3e294f77604db32ec